doc/getting-started.md

```diff
@@ -0,0 +1,344 @@
+3d2def20cd0d3aca148741ef469bda11647a3040d7669c82745d03c728706a8b
+
+# Get block data
+$ dogecoin-cli getblock 3d2def20cd0d3aca148741ef469bda11647a3040d7669c82745d03c728706a8b
+{
+  "hash": "3d2def20cd0d3aca148741ef469bda11647a3040d7669c82745d03c728706a8b",
+  "confirmations": 7816,
+  "strippedsize": 190,
+  "size": 190,
+  "weight": 760,
+  "height": 69,
+  "version": 1,
+  "versionHex": "00000001",
+  "merkleroot": "695ce4208fa7a87ef9e99805b0910dc129058ecdceb5cef7e25f71dcdc7936db",
+  "tx": [
+    "695ce4208fa7a87ef9e99805b0910dc129058ecdceb5cef7e25f71dcdc7936db"
+  ],
+  "time": 1386475225,
+  "mediantime": 1386475209,
+  "nonce": 3923708672,
+  "bits": "1e0ffff0",
+  "difficulty": 0.000244140625,
+  "chainwork": "0000000000000000000000000000000000000000000000000000000004600460",
+  "previousblockhash": "ffa69e04f928b84f19d84da25fb544340e54dca6c03c33930da245719e61c5ea",
+  "nextblockhash": "44bf8abbbb96d4dcfb95df563e606c37987133ea3e013b23bbddde8d7f905fdd"
+}
+```
+
+See the `tx` field to find the list of all block transactions. Only one exist in the block 69th, a coinbase transaction.
+
+To get transaction details :
+
+```bash
+$ dogecoin-cli getrawtransaction 695ce4208fa7a87ef9e99805b0910dc129058ecdceb5cef7e25f71dcdc7936db
+01000000010000000000000000000000000000000000000000000000000000000000000000ffffffff0e04d9eea3520101062f503253482fffffffff0100ac6156be23000023210340a42a5ad6c4c0cd5ae539657032e0a359bd3e0f95771f34d71691b13460a624ac00000000
+
+$ dogecoin-cli decoderawtransaction 01000000010000000000000000000000000000000000000000000000000000000000000000ffffffff0e04d9eea3520101062f503253482fffffffff0100ac6156be23000023210340a42a5ad6c4c0cd5ae539657032e0a359bd3e0f95771f34d71691b13460a624ac00000000
+{
+  "txid": "695ce4208fa7a87ef9e99805b0910dc129058ecdceb5cef7e25f71dcdc7936db",
+  "hash": "695ce4208fa7a87ef9e99805b0910dc129058ecdceb5cef7e25f71dcdc7936db",
+  "size": 109,
+  "vsize": 109,
+  "version": 1,
+  "locktime": 0,
+  "vin": [
+    {
+      "coinbase": "04d9eea3520101062f503253482f",
+      "sequence": 4294967295
+    }
+  ],
+  "vout": [
+    {
+      "value": 393004.00000000,
+      "n": 0,
+      "scriptPubKey": {
+        "asm": "0340a42a5ad6c4c0cd5ae539657032e0a359bd3e0f95771f34d71691b13460a624 OP_CHECKSIG",
+        "hex": "210340a42a5ad6c4c0cd5ae539657032e0a359bd3e0f95771f34d71691b13460a624ac",
+        "reqSigs": 1,
+        "type": "pubkey",
+        "addresses": [
+          "D5TjotnkoXekNQBCn54CBWxjEzCJMMe7WS"
+        ]
+      }
+    }
+  ]
+}
+```
+
+`vin` and `vout` will give you information about transaction inputs and outputs.
+
+## Node configuration
+
+A server contain a lot of parameter to configure, from memory limits, to RPC credentials, ports... There is two ways to prepare your perfect node !
+
+Using directly `dogecoind` options, `--help` will display all configuration parameters.
+
+**Command example :**
+```bash
+$ dogecoind -testnet -datadir=/path/to/dir -rpcuser=Shibetoshi -rpcpassword=Nakamoto
```

Comment:
would not add `-testnet` here, and also note that `-rpcuser` and `-rpcpassword` are not the recommended way to run a local node, but instead there is the auth cookie mechanism. Changing the datadir is dangerous too, so perhaps we can use a different example?

Reply:
> Changing the datadir is dangerous too.

I understand for credentials (related to bash history issues), but what are the risks for changing datadir ?

Why not adding `-testnet` here ? It's just to show a node with some configurations, testnet may be common, even more for devs.

Reply:
So it's not dangerous but time-consuming. Couldn't we rely on logic of users ? It's showing possibilities, and at worst it's something they can stop if they already synced the blockchain.

Reply:
Well, if you sent yourself coin to the wallet in datadir A and then you create a new wallet in datadir B, you better not overwrite those wallet.dat files from B to A...

It complicates things and shouldn't be in a beginner's guide imho. I guess you really feel this is important though? Why?

Reply:
I think it's important to be able to manage the directory where data are stored, whichever is the program.

And they have to modify it manually to make it works, `/path/to/dir` location will break the command.

Reply:
Alright. To be practical then: if you really insist on keeping it, could you please make a section about this and not introduce it in the example of passing arguments? Please also document the impact on using dogecoin-cli when having a custom datadir and/or rpc credentials there.

As for passing arguments, maybe we can make a case around `-paytxfee` and `-maxconnections` instead? Those are easier to comprehend too and don't impact other commands.

Reply:
Okay having configuration example of `dogecoind` & the configuration file with the following :

```
dogecoind -daemon -server -listen -paytxfee=0.01 -sendfreetransactions=1 -maxconnections=150
```
```
daemon=1
server=1
listen=1
paytxfee=0.01
sendfreetransactions=1
maxconnections=150
```
I'm not sure if `server` and `listen` are useful with `dogecoind`, seem default settings. Possibly you suggested [this config](https://github.com/dogecoin/dogecoin/pull/2568#discussion_r740535233) for `dogecoin-qt`.

Reply:
yeah you can remove the `server` and `listen`, keeps it shorter.

Reply:
Example replaced.